sdk/src/main/java/com/google/cloud/dataflow/sdk/util/BigQueryServicesImpl.java

```diff
@@ -244,6 +244,46 @@ public JobStatistics dryRunQuery(String projectId, String query)
           Sleeper.DEFAULT,
           backoff).getStatistics();
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Retries the RPC for at most {@code MAX_RPC_ATTEMPTS} times until it succeeds.
+     *
+     * @throws IOException if it exceeds max RPC retries.
+     */
+    @Override
+    public Job getJob(JobReference jobRef) throws IOException, InterruptedException {
+      BackOff backoff =
+          new AttemptBoundedExponentialBackOff(MAX_RPC_ATTEMPTS, INITIAL_RPC_BACKOFF_MILLIS);
+      return getJob(jobRef, Sleeper.DEFAULT, backoff);
+    }
+
+    @VisibleForTesting
+    public Job getJob(JobReference jobRef, Sleeper sleeper, BackOff backoff)
+        throws IOException, InterruptedException {
+      Exception lastException;
+      do {
+        try {
+          return client.jobs().get(jobRef.getProjectId(), jobRef.getJobId()).execute();
+        } catch (GoogleJsonResponseException e) {
+          if (errorExtractor.itemNotFound(e)) {
+            LOG.info("Job {} does not exists.", jobRef.getJobId());
+            return null;
+          }
+          LOG.warn("Ignore the error and retry getting the job.", e);
```

Comment:
("Ignoring error encountered while trying to query the status of BigQuery job {}", jobId, e)